### DISCUSSION TITLE
Workaround for providing support for EnumMemberAttribute

### DISCUSSION THREAD

**Discussion by snjosephms** [dotnet/runtime#80858] [Q&A]
We are migrating our code base from Newtonsoft to System.Text.Json.
Our data model uses the EnumMemberAttribute decorations https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.enummemberattribute?view=net-7.0.

Based on this post, I see that the support for this attribute is not provided in System.Text.Json out of the box : https://github.com/dotnet/runtime/issues/74385#issuecomment-1223931204

The post mentions that we can implement custom converters to provide support for this - but its unclear to me how this can be done. 
 Can you please provide a sample implementation of such a custom converter?

**Comment by ghost** [comment]
Tagging subscribers to this area: @dotnet/area-system-text-json, @gregsdennis
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.
<details>
<summary>Issue Details</summary>
<hr />

We are migrating our code base from Newtonsoft to System.Text.Json.
Our data model uses the EnumMemberAttribute decorations https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.enummemberattribute?view=net-7.0.

Based on this post, I see that the support for this attribute is not provided in System.Text.Json out of the box : https://github.com/dotnet/runtime/issues/74385#issuecomment-1223931204

I there any workaround we can implement to provide this support?

<table>
  <tr>
    <th align="left">Author:</th>
    <td>snjosephms</td>
  </tr>
  <tr>
    <th align="left">Assignees:</th>
    <td>-</td>
  </tr>
  <tr>
    <th align="left">Labels:</th>
    <td>

`area-System.Text.Json`

</td>
  </tr>
  <tr>
    <th align="left">Milestone:</th>
    <td>-</td>
  </tr>
</table>
</details>

**Comment by gregsdennis** [comment]
That attribute isn't part of JSON serialization.  That's for ASP.Net data contracts.

You can use `EnumStringConverter<T>` from my STJ utilities package [Json.More.Net](https://nuget.org/packages/Json.More.Net).  More info at https://json-everything.net (click on "Additional Utilities" at the right of the header bar).
